src/tools/compiletest/src/runtest.rs

```diff
@@ -1422,10 +1422,21 @@ impl<'test> TestCx<'test> {
     }
 
     fn compile_test(&self) -> ProcRes {
-        let mut rustc = self.make_compile_args(
-            &self.testpaths.file,
-            TargetLocation::ThisFile(self.make_exe_name()),
-        );
+        // Only use `make_exe_name` when the test ends up being executed.
```

Comment:
Where did these changes come from and/or how are they related to the PR's title?

Reply:
See the PR description - they remove the need for normalization, by fixing the `compiletest` bugs that affected #60006. IMO, the resulting tests are more clearly correct.

Reply:
Ah ok, sorry for missing that!